### PR TITLE
Ensure fate-sharing for the migration lock

### PR DESCRIPTION
**Pull request by JLockerman**
This commit switches migration to use the same connection that holds the migration-lock instead of a separate connection pool. Using a separate pool has a failure mode where the connection holding the lock dies, but the pool remains alive and keeps migrating, potentially allowing a race among multiple migrators. If the same connection is used, the connection dying will also force migration to stop.

### CODE REVIEW

File: pkg/pgmodel/end_to_end_tests/main_test.go
Comment (davidkohn88) [inline pkg/pgmodel/end_to_end_tests/main_test.go:113]: Do we need a separate pool? Isn't this always only going to be one connection?

File: pkg/runner/runner.go
Comment (cevian) [inline pkg/runner/runner.go:162]: schemaVersionLease allowed to be nil here, so this doesn't work

File: pkg/runner/runner.go
Comment (cevian) [inline pkg/runner/runner.go:307]: Should this also use the schemaLeaseLock connection for fate-sharing?

**Review by cevian — CHANGES_REQUESTED** [review]
Mostly looks good but one issue with incorrect nil handling

File: pkg/pgmodel/end_to_end_tests/main_test.go
Comment (davidkohn88) [reply pkg/pgmodel/end_to_end_tests/main_test.go:113]: Seems like maybe it is a single connection, in which case, the naming is the problem.

Comment (JLockerman) [reply pkg/pgmodel/end_to_end_tests/main_test.go:113]: we don't, but it's easier than refactoring the test more deeply

File: pkg/runner/runner.go
Comment (JLockerman) [reply pkg/runner/runner.go:162]: blah, forgot we allowed that 😞

File: pkg/runner/runner.go
Comment (JLockerman) [reply pkg/runner/runner.go:307]: it doesn't hurt, but it matters less: all the Client connections will perform their own checking, so this check is really just so we can print a nice error message in the event we have the wrong version; it's not a correctness hazard.

**Comment by JLockerman** [comment]
note: I want to wait until PR #262 is merged before this is, that way we'll have testing for the new behavior in CI 😋

File: pkg/runner/runner.go
Comment (JLockerman) [reply pkg/runner/runner.go]: it was easy so I changed it

**Review by cevian — APPROVED** [review]

**Pull request merged**
